auto_landing/src/pose.cpp

```diff
@@ -0,0 +1,67 @@
+#include <auto_landing/pose.hpp>
+
+namespace ariitk::auto_landing {
+
+PoseEstimation::PoseEstimation()
+    : height_(1) {}
+
+void PoseEstimation::init(ros::NodeHandle& nh , ros::NodeHandle& nh_private) {
+
+    set_firefly_pose_pub_ = nh.advertise<geometry_msgs::PoseStamped>("command_pose",1);
+    firefly_pose_sub_ = nh.subscribe("quad_odometry",1,&PoseEstimation::quadPoseCallBack,this);
+    firefly_pixel_coordinates_sub_ = nh.subscribe("pixel_coordinates",1,&PoseEstimation::pixelCoordinatesCallBack,this);
+
+    nh_private.getParam("K",camera_matrix_);
+    nh_private.getParam("P",projection_matrix_);
+    nh_private.getParam("R",rotation_matrix_);
```

Comment:
R is not a rotation matrix, it is the Rectification matrix. The rotation matrix of the camera frame w.r.t. the quad frame depends on how you have attached the camera to the drone.